birthday/birthday.py

```diff
@@ -0,0 +1,284 @@
+            while self == self.bot.get_cog(self.__class__.__name__):
+                now = datetime.datetime.utcnow()
+                tomorrow = (now + datetime.timedelta(days=1)).replace(hour=0, minute=0, second=0, microsecond=0)
+                await self.clean_yesterday_bdays()
+                await self.do_today_bdays()
+                await asyncio.sleep((tomorrow - now).total_seconds())
+
+    def cog_unload(self):
+        self.bday_loop.cancel()
+
+    # Commands
+    @commands.group()
+    @commands.guild_only()
+    async def bday(self, ctx: Context):
+        """Birthday settings"""
+        pass
+
+    @bday.command(name="channel")
+    @checks.mod_or_permissions(manage_roles=True)
+    async def bday_channel(self, ctx: Context, channel: discord.TextChannel):
+        """Sets the birthday announcement channel"""
+        message = ctx.message
+        guild = message.guild
+        await self.config.guild(channel.guild).channel.set(channel.id)
+        await message.channel.send(self.CHANNEL_SET(g=guild.name, c=channel.name))
+
+    @bday.command(name="role")
+    @checks.mod_or_permissions(manage_roles=True)
+    async def bday_role(self, ctx: Context, *, role: discord.Role):
+        """Sets the birthday role"""
+        message = ctx.message
+        guild = message.guild
+        await self.config.guild(role.guild).role.set(role.id)
+        await message.channel.send(self.ROLE_SET(g=guild.name, r=role.name))
+
+    @bday.command(name="remove", aliases=["del", "clear", "rm"])
+    async def bday_remove(self, ctx: Context):
+        """Unsets your birthday date"""
+        message = ctx.message
+        await self.remove_user_bday(message.guild.id, message.author.id)
+        await message.channel.send(self.BDAY_REMOVED())
+
+    @bday.command(name="set")
+    async def bday_set(self, ctx: Context, *, date: str):
+        """Sets your birthday date
+
+        The given date can either be month day, or day month
+        Year is optional. If not given, the age won't be displayed."""
+        message = ctx.message
+        channel = message.channel
+        author = message.author
+        year = None
+        birthday = self.parse_date(date)
+        # An Invalid date was entered.
+        if birthday is None:
+            print(self.BDAY_INVALID())
+            await channel.send(self.BDAY_INVALID())
+        else:
+            print(type(birthday))
+            if datetime.datetime.utcnow().year != birthday.year:
+                year = birthday.year
+            birthday = datetime.date(1, birthday.month, birthday.day)
+            await self.remove_user_bday(message.guild.id, author.id)
+            await self.get_date_config(message.guild.id, birthday.toordinal()).get_attr(author.id).set(year)
+            bday_month_str = birthday.strftime("%B")
+            bday_day_str = birthday.strftime("%d").lstrip("0")
+
+            await channel.send(self.BDAY_SET(bday_month_str + " " + bday_day_str))
```

Comment:
again, should be using ctx here, as this is redunant. Should also update all birthdays when someone sets their birthday incase they set it to the same day. (you'll need to modify your functions so it doesn't resend other birthday message.